activerecord/test/cases/associations/has_many_through_associations_test.rb

```diff
@@ -1252,4 +1255,56 @@ def test_incorrectly_ordered_through_associations
       )
     end
   end
+
+  def test_belongs_to_autosave_true_works_for_one_association
+    cs180 = Seminar.new(name: "CS180")
+    fall = Session.new(name: "Fall",
+                        start_date: "2017-09-01",
+                        end_date: "2017-12-31")
+    fall.sections << (cs180a = cs180.sections.build(short_name: "A"))
+
+    assert cs180.save
+
+    cs180.reload
+    fall.reload
+    assert_equal [cs180a].sort_by(&:id), cs180.sections.sort_by(&:id)
+    assert_equal [cs180a].sort_by(&:id), fall.sections.sort_by(&:id)
+  end
+
+  def test_belongs_to_autosave_true_fails_for_two_associations
```

Comment:
Does this test describe a failure? The test content seems like everything work.

Reply:
Do you want me to remove the other tests and leave only `test_belongs_to_autosave_true_fails_for_two_associations`?

Reply:
I would remove the other tests and rewrite this test to be something like this:
```ruby
def test_circular_autosave_association_correctly_saves_multiple_records
  cs180 = Seminar.new(name: "CS180")
  fall = Session.new(name: "Fall")
  sections = [
    cs180.sections.build(short_name: "A"),
    cs180.sections.build(short_name: "B"),
  ]
  fall.sections << sections
  fall.save!
  fall.reload
  assert_equal sections, fall.sections.sort_by(&:id)
end
```